lib/blocs/sync/sync_cubit.dart

```diff
@@ -698,6 +716,26 @@ class SyncCubit extends Cubit<SyncState> {
     });
   }
 
+  bool _isOverThePedingTime(DateTime? transactionCreatedDate) {
+    // If don't have the date information we cannot assume that is over the pending time
+    if (transactionCreatedDate == null) {
+      return false;
+    }
+
+    return DateTime.now().isAfter(transactionCreatedDate.add(_pendingWaitTime));
+  }
+
+  Future<DateTime?> _getDateCreatedByDataTx(String dataTx) async {
+    final rev = await _driveDao.fileRevisionByDataTex(tx: dataTx).get();
```

Comment:
This one is not actually part of your PR, but another spelling nit here:

```suggestion
    final rev = await _driveDao.fileRevisionByDataTx(tx: dataTx).get();
```